04-Bash-Scripting.md

```diff
@@ -698,3 +694,4 @@ In this book, we'll use `[[ ]]` because it has fewer gotchas.
 [^javascript2]: If you know some JavaScript you might be familiar with the problem of too-permissive operators:
 in JS, `"4" + 1 == "41"`, but `"4" - 1 == 3`.
 [^interview]: Also, why do so many people ask this as an interview question!?
+[^lexicographically]: I.e., how [`strcmp()`](http://www.cplusplus.com/reference/cstring/strcmp/) sorts strings; if `LHS < RHS`, then `strcmp(LHS, RHS) < 0`.
```

Comment:
`<`/`>` don't compare in the same manner as `strcomp` does for example if you have `strcmp("zebra", "pineapples")` you will rightfully get a return values of less than 0 but in bash string comparison has entirely different behavior. For example the following code will out put `pineapple is shorter than zebra` simply because "z" has a higher ascii value than "p".
```bash
if [[ "zebra" < "pineapple" ]]; then
    echo "zebra is shorter than pineapple."
else
    echo "Pineapple is shorter than zebra."
fi
```

Reply:
Oh, I think I have the condition backwards--it should be "if `LHS < RHS`, then `strcmp(LHS, RHS) > 0`?

Reply:
My mistake, what you have in your commit is correct.